I84 wave 1

PeterAlabaster opened the pull request:
Handle refused blood glucose observations seperately to refused observations.

Hopefully this also cross-covers any other places where a refused observation would be taken into account.

This has been done by adding a new column to spell, as opposed to storing it on the activity, due to the current data model not supporting storing/pulling the value from activity due to the way most of the data flow relies on nh.activity.spell, and spells having a o2m relation to activities.

Commit:
[I84] Add new column for blood glucose refusal.

Also add handling for said column UP UNTIL the column is effectively
re-mapped in the SQL view from `refusing_obs` to `refusal_in_effect`

Just want to make sure commits are seperate so I don't get confused.

Commit:
[I84] Minor syntax error

Commit:
[I84] Move logic to correct model, use correct data_model

DaeronAlagos commented on nh_eobs_mental_health/models/nh_clinical_pme_obs_stop.py:
```diff
@@ -45,7 +45,9 @@ def start(self, activity_id):
         self.set_obs_stop_flag(True)
-        self.set_refusing_obs_flag(False)
+        on_blood_glucose = activity.data_model == 'nh.clinical.patient.observation.blood_glucose'
+        import pdb;pdb.set_trace()
```
Oops!

DaeronAlagos commented on nh_eobs_mental_health/models/nh_clinical_wardboard.py:
```diff
@@ -319,6 +320,7 @@ def read(self, cr, user, ids, fields=None, context=None,
                 elif spell.get('refusing_obs'):
                     rec['frequency'] = 'Refused - {0}'.format(rec['frequency'])
                     rec['next_diff'] = 'Refused - {0}'.format(rec['next_diff'])
+                elif spell.get('refusing_obs_blood_glucose'):
```
May be missing from original code or not needed.

```suggestion
                elif spell.get('refusing_obs_blood_glucose'):
                    rec['bg_frequency'] = 'Refused - {0}'.format(rec['bg_frequency'])
```

DaeronAlagos reviewed (changes requested):
See inline comments

Commit:
[I84] Remove debugger

Commit:
[I84] Set bg next diff in new block. And rework from elif to if

It doesn't make sense to handle refusing general obs and refusing
bg obs as an if --> elif decision. If a patient is refusing
both, then both need to be displayed. Thus changing the elif to if
